### PR TITLE
KEP-2305: introduce allow-metric-labels-manifest

### PR DESCRIPTION
**Pull request by rexagod**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:
Enable configuring the allow-list by reading a manifest. This is done as a part of graduating https://github.com/kubernetes/enhancements/issues/2305 to [beta](https://github.com/kubernetes/enhancements/pull/4039/files#diff-3b7c51c078186aca805a75e2f844e82d9b4be550a37627f0c89ece1225145becR333).

#### Does this PR introduce a user-facing change?

```release-note
allow-list of metric labels can be configured by supplying a manifest using the --allow-metric-labels-manifest flag
```
#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:
```docs
- [KEP]: https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/2305-metrics-cardinality-enforcement
- [Usage]: --allow-metric-labels-manifest <path-to-the-manifest>
```

### CODE REVIEW

File: staging/src/k8s.io/component-base/metrics/opts.go [hunk 1]
```diff
@@ -18,11 +18,16 @@ package metrics
...
+	"os"
+	"path/filepath"
 	"strings"
 	"sync"
 	"time"
 
 	"github.com/prometheus/client_golang/prometheus"
+	"gopkg.in/yaml.v2"
+	"k8s.io/klog/v2"
+
```
Comment (logicalhan) [inline staging/src/k8s.io/component-base/metrics/opts.go:30]:
```suggestion

	"k8s.io/klog/v2"
```

**Review by logicalhan — APPROVED** [review]
/lgtm
/approve

**Comment by rexagod** [comment]
/assign @johnbelamaric 
(as per the comment above)

File: staging/src/k8s.io/component-base/metrics/opts.go [hunk 2]
```diff
@@ -354,3 +361,20 @@ func SetLabelAllowListFromCLI(allowListMapping map[string]string) {
...
+	defer allowListLock.Unlock()
+	allowListMapping := make(map[string]string)
+	data, err := os.ReadFile(filepath.Clean(manifest))
+	if err != nil {
+		klog.Errorf("Failed to read allow list manifest: %v", err)
+		return
+	}
+	err = yaml.Unmarshal(data, &allowListMapping)
+	if err != nil {
+		klog.Errorf("Failed to parse allow list manifest: %v", err)
```
Comment (johnbelamaric) [inline staging/src/k8s.io/component-base/metrics/opts.go:376]: Should there be error handling / panic or something here? Is logging sufficient? @logicalhan what is the expectation here?

Comment (logicalhan) [reply staging/src/k8s.io/component-base/metrics/opts.go:376]: Falling back to default behavior seems fine. I wouldn't want this to explode people.

**Comment by furkatgofurov7** [comment]
As per [slack discussion](https://kubernetes.slack.com/archives/C2C40FMNF/p1690208795626569?thread_ts=1689783786.018389&cid=C2C40FMNF) moving this out of the current milestone. 

/milestone next

**Review by logicalhan — APPROVED** [review]
/lgtm
/approve

**Pull request merged**